multiplexer/main.go

```diff
@@ -13,33 +19,135 @@ type URLs struct {
 	List []string `json:"list"`
 }
 
+//Client is a http clieng
+type Client struct {
+	client *http.Client
+}
+
+//Response is the response sent to the client
+type Response struct {
+	index   int
+	content []byte
+}
+
+//Job is a single url with its index
+type Job struct {
+	index int
+	site  string
+}
+
+func newClient() *Client {
+	return &Client{
+		client: &http.Client{
+			Timeout: time.Second,
+		},
+	}
+}
+
+func (c Client) getContent(url string) ([]byte, error) {
+	resp, err := c.client.Get(url)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	return b, nil
+}
+
 func main() {
 	var urls URLs
+	var jwg sync.WaitGroup
+	var rwg sync.WaitGroup
+	jobs := make(chan Job)
```

Comment:
Please make two channels buffered. As we know that max number of jobs will be ngt 20, you can allocate channels with 20 elements in each